lib/options.go

```diff
@@ -500,11 +500,38 @@ func (o Options) ForEachSpecified(structTag string, callback func(key string, va
 	for i := 0; i < structType.NumField(); i++ {
 		fieldType := structType.Field(i)
 		fieldVal := structVal.Field(i)
+		value := fieldVal.Interface()
 
 		shouldCall := false
 		switch fieldType.Type.Kind() {
 		case reflect.Struct:
+			// Unpack any guregu/null values
 			shouldCall = fieldVal.FieldByName("Valid").Bool()
+			valOrZero := fieldVal.MethodByName("ValueOrZero")
+			if shouldCall {
+				if valOrZero.IsValid() {
+					val := valOrZero.Call([]reflect.Value{})[0]
+					switch val.Kind() {
+					case reflect.Bool:
+						value = val.Bool()
+					case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+						value = val.Int()
+					case reflect.Uint, reflect.Uintptr, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+						value = val.Uint()
+					case reflect.Float32, reflect.Float64:
+						value = val.Float()
+					case reflect.String:
+						value = val.String()
+					default:
+						value = val.Interface()
+					}
```

Comment:
I'm not fond of the nested `switch` here, but didn't want to complicate things further.

Also, this is not an exhaustive `Kind` check, but it handles most of the `guregu` nullable types except `null.Time`, which we're not using.

Let me know if this can be improved.

Reply:
I don't think we need this `switch` at all, since we're already calling the `callback` with an `interface{}` value, we don't really care what the type of the `ValueOrZero()` result it. So, I think we can simply use the [`reflect/Value.Interface()` method](https://golang.org/pkg/reflect/#Value.Interface)...

Reply:
:man_facepalming: Thanks, will fix.